cartographer/sensor/compressed_point_cloud.cc

```diff
@@ -145,9 +145,16 @@ CompressedPointCloud::CompressedPointCloud(const PointCloud& point_cloud)
   CHECK_EQ(num_blocks, 0);
 }
 
-CompressedPointCloud::CompressedPointCloud(const std::vector<int32>& point_data,
-                                           size_t num_points)
-    : point_data_(point_data), num_points_(num_points) {}
+CompressedPointCloud::CompressedPointCloud(
+    const proto::CompressedPointCloud& proto) {
+  CHECK(proto.has_num_points()) << "The compressed point cloud is malformed.";
+  num_points_ = proto.num_points();
+  const size_t data_size = proto.point_data_size();
```

Comment:
`size_t` -> `int`, to match the generated protobuf code and the comparison below.